gematria/datasets/pipelines/compile_modules_lib.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import apache_beam as beam
+from rules_python.python.runfiles import runfiles
+import os
+
+from absl import logging
+
+from collections.abc import Iterable
+from collections.abc import Callable
+
+import subprocess
+
+from gematria.datasets.python import extract_bbs_from_obj
+
+
+def _get_llvm_binary_path(tool_name: str) -> str:
+  runfiles_env = runfiles.Create(os.environ)
+  assert runfiles_env is not None
```

Comment:
We might want to cache this part (to avoid any expensive operations like extraction or copying files).

Reply:
It will be cached by the initialization of the implementation of `beam.DoFn` given it's only called in `__init__` there. I'm not sure caching it anywhere else makes sense as we could have things running across multiple machines where a process like extraction might need to be performed once per machine.